### PR TITLE
Add SPI slave communication interface

### PR DESCRIPTION
**Pull request by dakopian-figure**

### CODE REVIEW

File: system.h [hunk 1]
```diff
@@ -93,7 +93,7 @@ class System {
         api.add_api_variable("t_i_correction", new const APIFloat(&actuator_.main_loop_.param_.torque_correction));
         api.add_api_variable("log", new APICallback(get_log, log));
         api.add_api_variable("messages_version", new APICallback([](){ return MOTOR_MESSAGES_VERSION; }, [](std::string s) {} ));
-        api.add_api_variable("usb_err", new APIUint32(&((USBCommunication *) &communication_)->usb_.error_count_));
+//        api.add_api_variable("usb_err", new APIUint32(&((USBCommunication *) &communication_)->usb_.error_count_));
```
Comment (dakopian-figure) [inline system.h]: @leemagnusson what do I do with this statement if there is no USBCommunication in the project?

File: peripheral/spi_mailbox.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  if(mailbox_id < kMaxMailboxId)
+  {
+    auto pool = mailbox_pool_table_[mailbox_id];
+
+    // If mailbox is supported
+    if(pool != NULL)
+    {
+      // Find a buffer with a matching Id and the oldest sequence number.
+      // TODO: Testing/setting of the "busy" flag needs to be inside of a critical section
```
Comment (leemagnusson) [inline peripheral/spi_mailbox.cpp:118]: My strategy has been to use interupt priorities where possible. You may consider disabling just your interrupt or using the basepri register to disable low priority registers

File: scripts/configure_gcc.mk [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # debug build?
 DEBUG = 1
 # optimization
-OPT = -Og -fsingle-precision-constant -O3 
+OPT = -O1 -fsingle-precision-constant #-O3
```
Comment (leemagnusson) [inline scripts/configure_gcc.mk]: Will need to do with optimization. You'll want to make sure the code still works with optimization on. Sometimes it affects the flags between interrupt contexts

File: system.h (hunk 1 above)
Comment (leemagnusson) [reply system.h]: I will move this.

**Review by leemagnusson — COMMENTED** [review]
This looks great. You will need to reenable optimization. I'm not reviewing in too much detail. But it does look well set up.

File: scripts/configure_gcc.mk (hunk 3 above)
Comment (dakopian-figure) [reply scripts/configure_gcc.mk]: Reverted optimization settings.

File: scripts/configure.mk [hunk 4]
```diff
@@ -42,10 +42,11 @@ $(SELF_DIR)../peripheral/stm32g4/startup_stm32g474xx.s
 
 # macros for gcc
 # AS defines
-AS_DEFS = 
+AS_DEFS =
 
 # C defines
 override C_DEFS +=  \
+-DNO_WATCHDOG \
```
Comment (leemagnusson) [inline scripts/configure.mk]: Need to remove this

**Review by leemagnusson — CHANGES_REQUESTED** [review]

File: scripts/configure.mk (hunk 4 above)
Comment (dakopian-figure) [reply scripts/configure.mk]: Good catch - thanks! Removed

**Review by leemagnusson — APPROVED** [review]

**Pull request merged**